openvdb_points/tools/PointDataGrid.h

```diff
@@ -360,6 +360,65 @@ class PointDataLeafNode : public tree::LeafNode<T, Log2Dim> {
         mAttributeSet->reorderAttributes(replacement);
     }
 
+    /// @brief Swap the underlying attribute set with the given @a attributeSet.
+    /// This leaf will assume ownership of the given attribute set. The new
+    /// attributes must have an equal descriptor. Additionally, if the new
+    /// attributes have different topology, then the voxel offset values must
+    /// be updated to reflect this new topology after a call to this method.
+    void swap(AttributeSet* attributeSet)
+    {
+        if (!attributeSet) {
+            OPENVDB_THROW(ValueError, "Cannot swap with a null attribute set");
+        }
+
+        if (mAttributeSet->descriptor() != attributeSet->descriptor()) {
+            OPENVDB_THROW(ValueError, "Attribute set descriptors are not equal.");
+        }
+
+        mAttributeSet.reset(attributeSet);
+    }
+
+    /// @brief Sets all of the voxel offset values on this leaf, from the given vector
+    /// of @a offsets. If @a updateValueMask is true, then the active value mask will
+    /// also be updated such that voxels with points are marked as active, and empty voxels
+    /// are marked as inactive.
+    void setOffsets(const std::vector<ValueType>& offsets, const bool updateValueMask = true)
+    {
+        if (offsets.size() != LeafNodeType::NUM_VALUES) {
+            OPENVDB_THROW(ValueError, "Offset vector size doesn't match number of voxels.")
+        }
+
+        for (size_t index = 0; index < offsets.size(); ++index) {
+            setOffsetOnly(index, offsets[index]);
+        }
+
+        if (updateValueMask)
+            this->updateValueMask();
```

Comment:
Just a quick formatting thing, the Coding Style says this:

* 19. You may leave off braces for simple, single line flow control statements.

Think we should put this one on one-line in order to leave off the braces.